enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/load_balancing/procedure/GetServersProcedureForSingleDC.java

```diff
@@ -0,0 +1,172 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.load_balancing.procedure;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.neo4j.causalclustering.core.CausalClusteringSettings;
+import org.neo4j.causalclustering.core.consensus.LeaderLocator;
+import org.neo4j.causalclustering.core.consensus.NoLeaderFoundException;
+import org.neo4j.causalclustering.discovery.CoreServerInfo;
+import org.neo4j.causalclustering.discovery.TopologyService;
+import org.neo4j.causalclustering.identity.MemberId;
+import org.neo4j.causalclustering.load_balancing.Endpoint;
+import org.neo4j.causalclustering.load_balancing.LoadBalancingProcessor;
+import org.neo4j.causalclustering.load_balancing.LoadBalancingResult;
+import org.neo4j.collection.RawIterator;
+import org.neo4j.helpers.AdvertisedSocketAddress;
+import org.neo4j.kernel.api.exceptions.ProcedureException;
+import org.neo4j.kernel.api.proc.CallableProcedure;
+import org.neo4j.kernel.api.proc.Context;
+import org.neo4j.kernel.api.proc.Neo4jTypes;
+import org.neo4j.kernel.api.proc.ProcedureSignature;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.logging.Log;
+import org.neo4j.logging.LogProvider;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Stream.concat;
+import static org.neo4j.causalclustering.core.CausalClusteringSettings.cluster_allow_reads_on_followers;
+import static org.neo4j.causalclustering.load_balancing.Util.asList;
+import static org.neo4j.causalclustering.load_balancing.Util.extractBoltAddress;
+import static org.neo4j.causalclustering.load_balancing.procedure.ParameterNames.CONTEXT;
+import static org.neo4j.causalclustering.load_balancing.procedure.ParameterNames.SERVERS;
+import static org.neo4j.causalclustering.load_balancing.procedure.ParameterNames.TTL;
+import static org.neo4j.causalclustering.load_balancing.procedure.ProcedureNames.GET_SERVERS_V1;
+import static org.neo4j.causalclustering.load_balancing.procedure.ProcedureNames.GET_SERVERS_V2;
+import static org.neo4j.kernel.api.proc.ProcedureSignature.procedureSignature;
+
+/**
+ * Returns endpoints and their capabilities.
+ *
+ * GetServersV2 extends upon V1 by allowing a client context consisting of
+ * key-value pairs to be supplied to and used by the concrete load
+ * balancing strategies.
+ */
+public class GetServersProcedureForSingleDC implements CallableProcedure
```

Comment:
Perhaps at some point in time the Legacy/V1 will no longer be supported, but until then it is in fact supported, and used by 1.1 and 1.2 java drivers for example. Naming it with legacy seems very premature and also not really helpful. V1/V2/... should be the way to keep track of this externally published interface.